pkg/agent/eventsync.go

```diff
@@ -132,11 +135,40 @@ func (es *EventSync) initCache() error {
 	return nil
 }
 
-func (es *EventSync) rebuildCache() {
+func (es *EventSync) RebuildCache() {
+	// SDB - NOTE : Do we need to pause jobs.
+	logger.Info("rebuild cache")
+
 	agent.cacheManager.Flush()
 	if err := es.initCache(); err != nil {
 		logger.WithError(err).Error("failed to rebuild cache")
 	}
+
+	agentInstance := agent.agentResourceManager.GetAgentResource()
+	agentDetails := util.GetAgentDetails(agentInstance)
+	if agentDetails == nil {
+		agentDetails = make(map[string]interface{})
+	}
+
+	cacheUpdateTime := int64(0)
+
+	// update cacheUpdateTime in agent resource
+	value, exits := agentDetails["cacheUpdateTime"]
+	if !exits {
+		// cache update time hasn't been established yet.  Update time to 7 days from current now time
+		currentTime := time.Now().Add(7 * 24 * time.Hour)
+		cacheUpdateTime = currentTime.UnixNano()
+	} else {
+		// if the cache update time already exists, update to 7 days from current cache update time
+		parsedTime, _ := time.Parse(time.RFC3339, value.(string))
+		parsedTime.Add(7 * 24 * time.Hour)
+		cacheUpdateTime = parsedTime.UnixNano()
+	}
+
+	// persist cacheUpdateTime
+	agentDetails["cacheUpdateTime"] = strconv.FormatInt(cacheUpdateTime, 10)
+	agent.apicClient.CreateSubResource(agentInstance.ResourceMeta, map[string]interface{}{definitions.XAgentDetails: agentDetails})
```

Comment:
Rather then getting the details from the, creating the map above if necessary, then adding to the map, you can use the util method to help.

```
agentInstance := agent.agentResourceManager.GetAgentResource()
util.SetAgentDetailsKey(agentInstance, "cacheUpdateTime", strconv.FormatInt(cacheUpdateTime, 10))
agent.apicClient.CreateSubResource(agentInstance.ResourceMeta,map[string]interface{}{definitions.XAgentDetails: util.GetAgentDetails(agentInstance})
```

Reply:
Maybe make a util method that gets just the details subres, returning this.

`map[string]interface{}{definitions.XAgentDetails: util.GetAgentDetails(agentInstance}`

Reply:
can do...